index.js

```diff
@@ -1,31 +1,39 @@
 const Client = require('./lib');
 const plugins = require('./plugins');
 
-// default plugins
-const {
-    Reconnection,
-    GracefulShutdown,
-    ConnectionRetry,
-    Duplex,
-    Encoding,
-    Retry
-} = plugins;
+const Plugins = {
+    gracefulShutdown: plugins.ConnectionRetry,
+    connectionRetry: plugins.ConnectionRetry,
+    reconnection: plugins.Reconnection,
+    duplex: plugins.Duplex,
+    encoding: plugins.Encoding,
+    rpc: plugins.RPC,
+    confirm: plugins.Confirm,
+    retry: plugins.Retry
+};
+
+const resolvePlugins = (plugins) => plugins
+    .map((plugin) => typeof plugin === 'string' ? new Plugins[plugin]() : plugin);
```

Comment:
Maybe you can assert that the plugin exists before trying to instantiate it, so the error would be clear for the lib user

Reply:
Yes it should be there, I believe.
Thanks.